sql/sql_table.cc

```diff
@@ -3876,8 +3876,19 @@ mysql_prepare_create_table(THD *thd, HA_CREATE_INFO *create_info,
             column->length= MAX_LEN_GEOM_POINT_FIELD;
 	  if (!column->length)
 	  {
-	    my_error(ER_BLOB_KEY_WITHOUT_LENGTH, MYF(0), column->field_name.str);
-	    DBUG_RETURN(TRUE);
+    /*currently Only InnoDB and MyISAM supports long unique constraints*/
+		  if (key->type == Key::UNIQUE && 
```

Comment:
Indention incorrect and again why you can't use here file->ht()->db_type == DB_TYPE_INNODB || file->ht()->db_type == DB_TYPE_MYISAM  here ?